src/app/src/test/kotlin/org/vaccineimpact/orderlyweb/tests/integration_tests/tests/web/WorkflowRunTests.kt

```diff
@@ -184,14 +158,41 @@ class WorkflowRunTests : IntegrationTest()
         )
 
         val orderlyServerResponse =
-                OrderlyServer(AppConfig()).get("/v1/workflow/${workflowRunResponse.key}/status/", emptyMap())
-        val workflowRunStatusResponse =
+                OrderlyServer(AppConfig()).get("/v1/workflow/$key/status/", emptyMap())
+        val orderlyWorkflowRunStatusResponse =
                 orderlyServerResponse.data(WorkflowRunController.WorkflowRunStatusResponse::class.java)
-        assertThat(workflowRunStatus.status).isEqualTo(workflowRunStatusResponse.status)
-        assertThat(workflowRunStatus.reports[0].name).isEqualTo("minimal")
-        assertThat(workflowRunStatus.reports[0].status).isEqualTo(workflowRunStatusResponse.reports[0].status)
-        assertThat(workflowRunStatus.reports[0].key).isEqualTo(workflowRunStatusResponse.reports[0].key)
-        assertThat(workflowRunStatus.reports[0].version).isEqualTo(workflowRunStatusResponse.reports[0].version)
+
+        assertThat(workflowRunStatus.status).isEqualTo(orderlyWorkflowRunStatusResponse.status)
+
+        var report = workflowRunStatus.reports[0]
+        var orderlyServerStatus = orderlyWorkflowRunStatusResponse.reports.find { it.key == report.key }!!
+        assertThat(report.name).isEqualTo("other")
+        assertThat(report.status).isEqualTo(orderlyServerStatus.status)
+        assertThat(report.key).isEqualTo(orderlyServerStatus.key)
+        assertThat(report.version).isEqualTo(orderlyServerStatus.version)
+
+        // these were queued in order: other, minimal, other, global
+        // but should now be ordered other, other, minimal, global because that will  be their execution order
```

Comment:
I'm not really sure why this ordering holds (don't think there are any dependencies involved), but I guess it's fine as long as this is stable behaviour from orderly server.

Reply:
Yeah this is because it will group multiple instances of the same report, nothing to do with dependencies. Tbh this test is a little brittle/tightly coupled to the orderly server functionality for my liking, I guess a better alternative would be to read the execution order from the submission response and make sure it's being respected in the final result.

Reply:
(although that logic is also already under test in the unit test!)

Reply:
> I guess a better alternative would be to read the execution order from the submission response

Actually the report names don't come back in the original submission response (which in itself feels a bit odd?) so we can't really use that